pandas/tests/frame/test_quantile.py

```diff
@@ -467,3 +467,14 @@ def test_quantile_empty(self):
 
         # FIXME (gives NaNs instead of NaT in 0.18.1 or 0.19.0)
         # res = df.quantile(0.5, numeric_only=False)
+
+    def test_quantile_empty_no_columns(self):
+        # GH#23925 _get_numeric_data may drop all columns
+        df = pd.DataFrame(pd.date_range("1/1/18", periods=5))
+        result = df.quantile(0.5)
+        expected = pd.Series([], index=[], name=0.5)
+        tm.assert_series_equal(result, expected)
+
+        result = df.quantile([0.5])
+        expected = pd.DataFrame([], index=[], columns=[0.5])
```

Comment:
I don't quite understand this expected. Should `index` and `columns` be flipped (and in the implementation too)?

With `DataFrame.quantlie([0.5])` the

* index is the `q` (`[0.5]`)
* columns are the (numeric) columns

Reply:
I think you're right, yah.  will update

Reply:
what do we do in the scalar-q case then?

Reply:
I think

```
In [11]: pd.Series(name=0.5)
Out[11]: Series([], Name: 0.5, dtype: float64)
```

right? So that it's equivalent to

```python
In [18]: pd.DataFrame({"A": [1]}).quantile().drop("A")
Out[18]: Series([], Name: 0.5, dtype: float64)
```